File: backend/app/models/incident.py
class Incident:
    def __init__(self, id, title, description, severity, status,
                 reporter_name, analyst_name, created_at, updated_at):
        self.id = id
        self.title = title
        self.description = description
        self.severity = severity
        self.status = status
        self.reporter_name = reporter_name
        self.analyst_name = analyst_name
        self.created_at = created_at
        self.updated_at = updated_at

    def to_dict(self):
        return {
            'id': self.id,
            'title': self.title,
            'description': self.description,
            'severity': self.severity,
            'status': self.status,
            'reporter_name': self.reporter_name,
            'analyst_name': self.analyst_name,
            'created_at': self.created_at,
            'updated_at': self.updated_at
        }

File: client_app/lib/models/incident.dart
class Incident {
  final int id;
  final String title;
  final String description;
  final String severity;
  final String status;
  final String reporterName;
  final String? analystName;
  final String createdAt;
  final String updatedAt;

  Incident({
    required this.id,
    required this.title,
    required this.description,
    required this.severity,
    required this.status,
    required this.reporterName,
    this.analystName,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Incident.fromJson(Map<String, dynamic> json) {
    return Incident(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      severity: json['severity'],
      status: json['status'],
      reporterName: json['reporter_name'],
      analystName: json['analyst_name'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }
}